tests/kubernetes-plugins/full.bats

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env bats
+
+load "$HELPERS_ROOT/test-helpers.bash"
+
+ensure_variables_set BATS_SUPPORT_ROOT BATS_ASSERT_ROOT BATS_DETIK_ROOT BATS_FILE_ROOT TEST_NAMESPACE FLUENTBIT_IMAGE_REPOSITORY FLUENTBIT_IMAGE_TAG
+
+load "$BATS_DETIK_ROOT/utils.bash"
+load "$BATS_DETIK_ROOT/linter.bash"
+load "$BATS_DETIK_ROOT/detik.bash"
+load "$BATS_SUPPORT_ROOT/load.bash"
+load "$BATS_ASSERT_ROOT/load.bash"
+load "$BATS_FILE_ROOT/load.bash"
+
+# These are required for bats-detik
+# shellcheck disable=SC2034
+DETIK_CLIENT_NAME="kubectl -n $TEST_NAMESPACE"
+# shellcheck disable=SC2034
+DETIK_CLIENT_NAMESPACE="${TEST_NAMESPACE}"
+FLUENTBIT_POD_NAME=""
+TEST_POD_NAME=""
+
+setup_file() {
+    # HELM_VALUES_EXTRA_FILE is a default file containing global helm
+    # options that can be optionally applied on helm install/upgrade
+    # by the test. This will fall back to $TEST_ROOT/defaults/values.yaml.tpl
+    # if not passed.
+    if [ -e  "${HELM_VALUES_EXTRA_FILE}" ]; then
+      envsubst < "${HELM_VALUES_EXTRA_FILE}" > "${HELM_VALUES_EXTRA_FILE%.*}"
+      export HELM_VALUES_EXTRA_FILE="${HELM_VALUES_EXTRA_FILE%.*}"
+    fi
+
+    # First check that we should run these conditional tests at all
+    run docker run --rm -t $FLUENTBIT_IMAGE_REPOSITORY:$FLUENTBIT_IMAGE_TAG /fluent-bit/bin/fluent-bit -F kubernetes --help
+    if [[ "$output" != *"namespace_labels"* ]]; then
```

Comment:
Let's move this to `setup` so it definitely executes on each test and not just once per file.

I don't think we should use `setup_file` unless it is super expensive set up, which this is not.

Reply:
the reason I have this in `setup_file` vs `setup` is that it prevents the rest of setup_file from running (such as setting up fluent-bit, which does take a longer amount of time). I can still move it, if it's preferred but we're adding an extra minute to test runs where fluent-bit doesn't need to be setup at all because it will be skipped

Reply:
Yeah I meant move the whole thing to `setup` rather than just this bit. I think that's acceptable as it's not obvious that `setup_file` will attempt to run again if it is skipped the first time.

Reply:
### setup_file vs setup

I made the changes you suggested and tested locally with `time FLUENTBIT_IMAGE_REPOSITORY=ghcr.io/fluent/fluent-bit/pr-8279 FLUENTBIT_IMAGE_TAG=18e5eda4b644723fcfbe6a46524de8430f856fe5-debug ./run-tests.sh tests/kubernetes-plugins/full.bats`, differences are: 

- `setup_file` only will run only once when there are multiple tests in the file. When using `setup_file` to ensure we should run all of the tests (the namespace_labels check), it's only installing fluent-bit once for the 3 included tests and the total time it took was `59.464 total` 

- if I merge `setup_file` (and `teardown_file`) logic into `setup`/`teardown` this has to run for every single test. the total time is `3:09.32 total` 

So it's ~2 minutes extra to merge this down into `setup`/`teardown`, I'd prefer to leave it as is.

---
### imagePullPolicy

- Our setup defaults to `imagePullPolicy: Always` within `defaults/values.yaml.tpl` which actually gets the default value from `./run-tests.sh:export HELM_IMAGE_PULL_POLICY=${HELM_IMAGE_PULL_POLICY:-Always}` so it will always pull unless we'd think about changing this to `IfNotPresent` globally?

- even if we set `imagePullPolicy: IfNotPresent` globally, we're creating a 3 node cluster, so it's only cached based on the node you get scheduled on, so we'd probably need something like a way to run a daemonset test first, that would cache the image everywhere? (i think chunk-rollover is currently first, and it is a daemsonset deployment, but would want to ensure we don't add another test before this without thinking - but maybe that's not that big of a deal?)

Reply:
OK, let's leave setup_file - it just looks smelly to me but I'll get over it :)

We have hit issues with imagePullPolicy defaults in the past which is why it is present so I don't really want to change it, e.g. re-running the tests on PR update was caching old images.

Really we should test by digest rather than tag but the helm chart did not support that at the time.

Let's leave that out of scope for now I think, I'd rather not have multiple changes in the same PR.